File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int insert = 0;

        for (int x : nums) {
            if (x != 0) {
                nums[insert++] = x;
            }
        }

        while (insert < nums.size()) {
            nums[insert++] = 0;
        }
    }
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int insert = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0 ) {
                int temp = nums[insert];
                nums[insert] = nums[i];
                nums[i] = temp;
                insert++;
            }
        }
    }
}